src/ansys/hps/client/common/base_resource.py

```diff
@@ -101,3 +101,12 @@ def __str__(self):
             dict_repr[key] = value
 
         return json.dumps(dict_repr, indent=2)
+
+    def __getitem__(self, key):
+        return getattr(self, key)
+
+    def __setitem__(self, key, value):
+        return setattr(self, key, value)
+
+    def get(self, key, default_value):
```

Comment:
Let's follow the [dict.get signature](https://docs.python.org/3/library/stdtypes.html#dict.get): `default_value` should be defaulted to None and called `default`.

Reply:
Changed as per review comment